docs/DeveloperGuide.md

```diff
@@ -1,38 +1,288 @@
 # Developer Guide
 
 ## Acknowledgements
-
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+ChatParser structure inspired by: 
+[this repository](https://github.com/kennethSty/ip)
+Calorie consumption bar inspired by 
+[this blogpost](https://medium.com/javarevisited/how-to-display-progressbar-on-the-standard-console-using-java-18f01d52b30e)
 
 ## Design & implementation
+### High Level Class Design
+The main classes of this implementation are:
+- HealthMate
+- ChatParser
+- User
+- MealList
+- MealEntriesList
+- Meal
+- MealEntry
+- HistoryTracker
+- UI
+
+In each class we focused on maintaining a tight abstraction barrier between classes.
+This specifically includes adherance to the "Tell Don't Ask" principle which was enforced by
+making most attributes of all classes above private and avoiding getter methods if possible.
+The following diagram illustrates the resulting associations, methods and attributes.
+For the sake of clarity, UI class is omitted since it is used across most classes to systematize any form of
+printing output to the user.
+
+![High Level CD](images/highLevelClassDiagram.jpg)
+
+#### HealthMate
+Entry point to the application is the main function of HealthMate.
+The HealthMate class contains a private ChatParser attribute. This attribute's run function initiates, 
+after an initial greeting to the user, the interaction process. In this process, 
+the user enters commands with additional information into his command line application.
+The content of these commands is parsed by the ChatParser. 
+
+#### ChatParser
+The ChatParser class, instantiated once per application run, manages HealthMates overal usage 
+flow through its main run() method. 
+
+It has two primary attributes: 
+- A `MealEntriesList` object called `mealEntries` 
+  - Contains tracked calorie consumption 
+- A `MealList` object called `mealOptions` 
+  - Contains meals that are presaved by the user for quick selection to track commonly consumed meals
+  - the command `meal menu` is used to display the current mealOptions. The implementation of this command is shown in the UML diagram below.
+![Meal Menu SD](images/mealMenuSD.png)
```

Comment:
perhaps the diagram could include a user so that the run() function does not appear out of nowhere, providing additional clarity.